### PR TITLE
Rel-507459 update applicationinstallhelper with rest

### PR DESCRIPTION
**Pull request by TommyBigNose**

### CODE REVIEW

File: CSharp/DevVmPowershell/DevVmPsModules/Cmdlets/AddApplicationFromLibraryModule.cs [hunk 1]
```diff
@@ -75,10 +75,15 @@ protected override void ProcessRecordCode()
 				relativityAdminPassword: RelativityAdminPassword,
 				sqlAdminUserName: SqlAdminUserName,
 				sqlAdminPassword: SqlAdminPassword);
-			IApplicationInstallHelper applicationInstallHelper = new ApplicationInstallHelper(connectionHelper);
+			ISqlRunner sqlRunner = new SqlRunner(connectionHelper);
+			ISqlHelper sqlHelper = new SqlHelper(sqlRunner);
+			RestHelper restHelper = new RestHelper();
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/DevVmPsModules/Cmdlets/AddApplicationFromLibraryModule.cs:80]: - [ ] Aaron created an interface IRestHelper, Can we use that instead to pass to the various Helper constructors?

File: CSharp/DevVmPowershell/Helpers/Implementations/ApplicationInstallHelper.cs [hunk 2]
```diff
@@ -1,173 +1,165 @@
...
+		}
+
+		private async Task<bool> DoesLibraryApplicationExistAsync(string applicationGuid)
+		{
+			List<LibraryApplicationResponse> allApps = await ReadAllLibraryApplicationAsync();
+
+			return allApps.Exists(x => x.Guids.Contains(new Guid(applicationGuid)));
+		}
+
+		private async Task<int> GetApplicationLibraryId(string applicationGuid)
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/ApplicationInstallHelper.cs:118]: - [ ] Add Async suffix

File: CSharp/DevVmPowershell/Helpers/Implementations/ApplicationInstallHelper.cs [hunk 3]
```diff
@@ -1,173 +1,165 @@
...
+			if (!await DoesLibraryApplicationExistAsync(applicationGuid))
+			{
+				throw new ValidationException("Library application does not exist");
+			}
+
+			List<LibraryApplicationResponse> allApps = await ReadAllLibraryApplicationAsync();
+			return allApps.Find(x => x.Guids.Contains(new Guid(applicationGuid))).ArtifactID;
+		}
+
+		private async Task<HttpResponseMessage> UpdateLibraryApplication(HttpClient httpClient, string filePath)
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/ApplicationInstallHelper.cs:129]: - [ ] Should the method name be Upload instead?

Comment (calimetikcura) [reply CSharp/DevVmPowershell/Helpers/Implementations/ApplicationInstallHelper.cs:129]: - [ ] Add Async suffix

File: CSharp/DevVmPowershell/Helpers/Implementations/ApplicationInstallHelper.cs [hunk 4]
```diff
@@ -1,173 +1,165 @@
...
+			form.Add(RestHelper.CreateFileContent(fileStream, fullFileName, contentType), Constants.Connection.RestUrlEndpoints.ApplicationInstall.uploadFileKeyName, fullFileName);
+			form.Headers.Add("X-CSRF-Header", "-");
+			form.Add(new StringContent(updateJsonRequest, Encoding.Unicode, "application/json"), "request");
+
+
+			HttpResponseMessage updateResponse = await RestHelper.MakePutAsync(httpClient, Constants.Connection.RestUrlEndpoints.ApplicationInstall.updateLibraryApplicationUrl, form);
+			return updateResponse;
+		}
+
+		private async Task<bool> InstallApplicationOnWorkspace(HttpClient httpClient, string workspaceName, int applicationId)
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/ApplicationInstallHelper.cs:151]: - [ ] Add Async suffix

File: CSharp/DevVmPowershell/Helpers/Implementations/RetryLogicHelper.cs [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+		/// <summary>
+		/// Basic validation that every function here uses
+		/// </summary>
+		/// <param name="numberOfRetries"></param>
+		/// <param name="delayInSeconds"></param>
+		/// <param name="operation"></param>
+		private void ValidateRetryParameters(int numberOfRetries, int delayInSeconds, object operation)
+		{
+			if (numberOfRetries < 0) throw new ArgumentException($"Parameter {nameof(numberOfRetries)} cannot be null or less than 0.");
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/RetryLogicHelper.cs:127]: - [ ] remove cannot be null from the message

File: CSharp/DevVmPowershell/Helpers/Implementations/RetryLogicHelper.cs [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+			if (delayInSeconds < 0) throw new ArgumentException($"Parameter {nameof(delayInSeconds)} cannot be null or less than 0.");
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/RetryLogicHelper.cs:128]: - [ ] remove cannot be null from the message

**Review by calimetikcura — CHANGES_REQUESTED** [review]

**Review by calimetikcura — APPROVED** [review]

**Review by aarongilbert123 — APPROVED** [review]

**Pull request merged**